### PR TITLE
Add IVS API to install latest prerelease package

### PR DESCRIPTION
**Pull request by joelverhagen**
https://github.com/NuGet/Home/issues/2148

See `IVsPackageInstaller2.cs` for the method signature.

@emgarten @yishaigalatzer @CyrusNajmabadi @zhili1208

### CODE REVIEW

File: test/TestExtensions/API.Test/InternalAPITestHook.cs [hunk 1]
```diff
@@ -6,6 +7,18 @@ namespace API.Test
 {
     public static class InternalAPITestHook
     {
+        public static void InstallLatestPackageApi(string id, bool prerelease)
+        {
+            var dte = ServiceLocator.GetInstance<EnvDTE.DTE>();
+            var services = ServiceLocator.GetInstance<IVsPackageInstaller2>();
```
Comment (alpaix) [inline test/TestExtensions/API.Test/InternalAPITestHook.cs:13]: Can it be null?

File: src/NuGet.Clients/NuGet.VisualStudio.Implementation/PreinstalledPackageInstaller.cs [hunk 2]
```diff
@@ -213,7 +213,7 @@ internal string GetRegistryRepositoryPath(string keyName, IEnumerable<IRegistryK
                         projectContext.PackageExtractionContext.UseLegacyPackageInstallPath = true;
 
                         // This runs from the UI thread
-                        await _installer.InstallInternalAsync(project, toInstall, repos, projectContext, package.IgnoreDependencies, CancellationToken.None);
+                        await _installer.InstallInternalAsync(project, toInstall, repos, projectContext, false, package.IgnoreDependencies, CancellationToken.None);
```
Comment (alpaix) [inline src/NuGet.Clients/NuGet.VisualStudio.Implementation/PreinstalledPackageInstaller.cs]: named parameter

File: test/TestExtensions/API.Test/InternalAPITestHook.cs (hunk 1 above)
Comment (emgarten) [inline test/TestExtensions/API.Test/InternalAPITestHook.cs:13]: Only if NuGet is not installed, it should be assumed that this exists.

Comment (alpaix) [inline test/TestExtensions/API.Test/InternalAPITestHook.cs:13]: or it's an earlier version that doesn't provide this service yet.
Given the scope of e2e tests I guess we will never have incompatible versions.

Comment (joelverhagen) [inline test/TestExtensions/API.Test/InternalAPITestHook.cs:13]: Correct. This is operating on the bits that were just built. (if not maybe we should consider the value of the tests... :wink:)

File: src/NuGet.Clients/NuGet.VisualStudio.Implementation/PreinstalledPackageInstaller.cs (hunk 2 above)
Comment (joelverhagen) [inline src/NuGet.Clients/NuGet.VisualStudio.Implementation/PreinstalledPackageInstaller.cs]: thanks, i missed this one.

Comment (joelverhagen) [inline src/NuGet.Clients/NuGet.VisualStudio.Implementation/PreinstalledPackageInstaller.cs]: Fixed.

**Comment by alpaix** [comment]
:shipit:

File: src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPackageInstaller.cs [hunk 3]
```diff
@@ -24,7 +24,8 @@
 namespace NuGet.VisualStudio
 {
     [Export(typeof(IVsPackageInstaller))]
-    public class VsPackageInstaller : IVsPackageInstaller
+    [Export(typeof(IVsPackageInstaller2))]
```
Comment (emgarten) [inline src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPackageInstaller.cs:27]: Typically in VS only the lowest interface is exported.

File: src/NuGet.Clients/VisualStudio/Extensibility/IVsPackageInstaller2.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        /// The package source to install the package from. This value can be <c>null</c>
+        /// to indicate that the user's configured sources should be used. Otherwise,
+        /// this should be the source path as a string. If the user has credentials
+        /// configured for a source, this value must exactly match the configured source
+        /// value.
+        /// </param>
+        /// <param name="project">The target project for package installation.</param>
+        /// <param name="packageId">The package ID of the package to install.</param>
+        /// <param name="includePrerelease">
+        /// Whether or to consider the prerelease version when finding the latest verion
```
Comment (emgarten) [inline src/NuGet.Clients/VisualStudio/Extensibility/IVsPackageInstaller2.cs]: typos

Comment (joelverhagen) [inline src/NuGet.Clients/VisualStudio/Extensibility/IVsPackageInstaller2.cs]: how do i english

File: src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPackageInstaller.cs (hunk 3 above)
Comment (joelverhagen) [inline src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPackageInstaller.cs:27]: Spoke offline to Justin about this. Exporting both allows all configurations:
1. Old caller can still get `IVsPackageInstaller`.
2. New caller can still get `IVsPackageInstaller` and try to cast up with `as`.
3. New caller can get `IVsPackageInstaller2` to require the new functionality.

**Comment by emgarten** [comment]
:rocket:

**Pull request merged**